server/command.go

```diff
@@ -231,6 +231,15 @@ func (p *Plugin) ExecuteCommand(c *plugin.Context, args *model.CommandArgs) (
 	*model.AppError) {
 	command, action, parameters := parseCommandArgs(args)
 
+	user, err := p.API.GetUser(args.UserId)
+	if err != nil {
+		return p.sendResponse(args, "User is not registered"), nil
+	}
+
+	if !user.IsSystemAdmin() {
+		return p.sendResponse(args, "Nagios command only for System Admin and Team Admin"), nil
+	}
```

Comment:
In the user is not a sysadmin, we need to check if the user is a team admin of the current team (taken from `args.TeamId`), using `p.API.HasPermissionToTeam`

Reply:
will update and push this one.